api/tests/routes/pro/patch_collective_offer_test.py

```diff
@@ -379,6 +380,74 @@ def when_user_is_not_attached_to_offerer(self, app, client):
         ]
         assert CollectiveOffer.query.get(offer.id).name == "Old name"
 
+    def test_cannot_update_offer_with_used_booking(self, client):
+        # Given
+        offer = CollectiveOfferFactory()
+        UsedCollectiveBookingFactory(
+            collectiveStock__collectiveOffer=offer,
+        )
+        offerers_factories.UserOffererFactory(
+            user__email="user@example.com",
+            offerer=offer.venue.managingOfferer,
+        )
+
+        # When
+        data = {
+            "name": "New name",
+        }
+        response = client.with_session_auth("user@example.com").patch(
+            f"/collective/offers/{humanize(offer.id)}", json=data
+        )
+
+        # Then
+        assert response.status_code == 403
+        assert response.json == {"offer": "Les offres utilisées ou remboursées ne sont pas modifiable."}
+
+    def test_patch_collective_offer_replacing_by_non_unknown_venue(self, client):
+        # Given
+        offerer = offerers_factories.OffererFactory()
+        offerers_factories.UserOffererFactory(
+            user__email="user@example.com",
+            offerer=offerer,
+        )
+        offer = CollectiveOfferFactory(venue__managingOfferer=offerer)
+
+        # When
+        data = {"venueId": 0}
+        response = client.with_session_auth("user@example.com").patch(
+            f"/collective/offers/{humanize(offer.id)}", json=data
+        )
+
+        # Then
+        assert response.status_code == 404
+        assert response.json["venueId"] == "The venue need to exist."
+
+    def test_patch_collective_offer_replacing_venue_with_different_offerer(self, client):
+        # Given
+        offerer = offerers_factories.OffererFactory()
+        offerer2 = offerers_factories.OffererFactory()
+        offerers_factories.UserOffererFactory(
+            user__email="user@example.com",
+            offerer=offerer,
+        )
+        offer = CollectiveOfferFactory(
+            venue__managingOfferer=offerer
+            # offer.venueId= 17442
+        )
+        venue2 = offerers_factories.VenueFactory(
+            managingOfferer=offerer2,
+        )
```

Comment:
Tu peux éviter le retour à la ligne appliqué par Black en supprimant la virgule à la fin de la liste des arguments de la fonction. C'est plus concis et ça reste facile à lire quand il y a peu d'argument comme ici.

```suggestion
        venue2 = offerers_factories.VenueFactory(managingOfferer=offerer2)
```

idem au-dessus.